### DISCUSSION TITLE
Jamulus server not showing ping and unable to connect

### DISCUSSION THREAD

**Discussion by RVLareu** [jamulussoftware/jamulus#1777] [Technical Support & Help]
Dear forum,

im having trouble while connecting to a certain server. The only way to see the server is if i  run Jamulus with the "--showallservers" flag. Once there, the ping for that server only shows eventually and its in only that moment that i can connect to it. I wasnt able to find any solution to this trouble. Would appreciate any suggestions.

System: windows 10. Tried with both ASIO4ALL versions and keeps happening.

Thanks

**Comment by dcorson-ticino-com** [comment]
Hi RV,
I would suggest that as it is only a certain server, that server is having big connection problems.
If it doesn't show up without showallservers your client is not receiving all the messages it should be receiving from the server.
It is unlikely that the server is only sending some messages.  More likely is that the internet conection between you and the server is so poor that many packets are being lost.  You can use a utility like pingplotter to follow packets between you and the server and see if and where they are being lost.

**Reply by howasparty** [reply]
This happens to me with many servers and does not depend on the OS I use.
I could not find any reason, as the servers I can't connect are in totally different locations. Traceroute is working to most of them. Other musicians I play with often change to servers I can't connect (and only see with the "--showallservers" flag).

**Reply by dcorson-ticino-com** [reply]
That is a certain sign that you are suffering from communication problems on your Internet connection.
Normally only the servers that respond are shown on the Connect list.
With --showallservers all the servers are shown that the Directory server mentions, even if they don't respond.
So you are not getting responses from many servers.  They are either not receiving the requests and thus never sending a response or the responses are not arriving.
I would suggest you contact your provider (or look for a different one).

**Reply by howasparty** [reply]
Thanks for your answer!

I must admit, it is not many servers, but some of them (see jpg)
![Bildschirmfoto 2021-06-15 um 14 42 48](https://user-images.githubusercontent.com/85933514/122054586-18f8d700-cde8-11eb-842c-cde490c10628.png)

As I wrote, Traceroute is working on most of them, which are not working. Ping times on other servers within my regions are short and making music on the servers that are working was fine with less problems than some other users.
So I wonder about the exact cause of this "communication problem", which might help me, to solve the problem.
But my ISP support seems to know less than me about network communication. Changing my ISP is no option at the moment.

**Reply by RVLareu** [reply]
it seems like a problem with the internet connection. After several tries the only workaround i found is that connecting to the servers first with mobile hotspot data, and then reconnecting with Wifi works like a charm.

**Reply by howasparty** [reply]
Wow!
Thanks, I will try it this way. Jamulus stays open while changing the networks, I guess?

**Reply by RVLareu** [reply]
yes, but you have to manually disconnect from the server you made the connection with while using mobile hotspot data, if not it stays there but not really connected. After that, in my case, with a Wifi connection i am able to see the Ping of the server (not sure at all about whats happening but its the only way i found that let me connect to certain server)

**Reply by dcorson-ticino-com** [reply]
Remember that Jamulus uses UDP.  Network tracing using TCP may not be giving the results you need.

**Reply by gene96817** [reply]
Ping and traceroute uses UDP messages. It could be interesting to run "ping - n 50" with your other parameters. That should make 50 ping test and at the end give you % of lost pings and the min/ave/max/standard dev on the ping times. Since you are having problems, you will probably be shocked by the results. 

> it seems like a problem with the internet connection. After several tries the only workaround i found is that connecting to the servers first with mobile hotspot data, and then reconnecting with Wifi works like a charm.

If this is consistent, the only way this makes sense is either (1) on startup some packets are lost and once it goes through ... or (2) your device as not properly configured for UDP packets. I hope it is not the case that your mobile hotspot firmware or configuration is mishandling initial UDP packets.
